src/XrdTpc/XrdTpcTPC.cc

```diff
@@ -952,6 +954,38 @@ int TPCHandler::ProcessPullReq(const std::string &resource, XrdHttpExtReq &req)
             logTransferEvent(LogMask::Error, rec, "PULL_FAIL", msg);
             return req.SendSimpleResp(rec.status, NULL, NULL, msg, 0);
     }
+    // ddavila 2023-01-05:
+    // The following change was required by the Rucio/SENSE project where
+    // multiple IP addresses, each from a different subnet, are assigned to a
+    // single server and routed differently by SENSE.
+    // The above requires the server to utilize the same IP, that was used to
+    // start the TPC, for the resolution of the given TPC instead of
+    // using any of the IPs available.
+    if (m_sense  == true ){
```

Comment:
I wonder if a more generic name isn't more suitable here, than using the name of the software that needs this (as it may also be useful in other domains, like within containerized setups). I thought maybe `tpc.route` with options `fixed` and `dynamic` (default), or something similar to that. For example, within Kubernetes, that may turn out to be needed as well.